core/src/impl/Kokkos_CPUDiscovery.hpp

```diff
@@ -18,6 +18,8 @@ namespace Impl {
 
 int mpi_ranks_per_node();
 int mpi_local_rank_on_node();
+// returns true if MPI execution environment is detected, false otherwise.
+bool is_mpi_exec();
```

Comment:
Personal opinion: a more expressive name that would not even need the comment is:
```cpp
bool mpi_detected()
```